### DISCUSSION TITLE
[开发]如何在修改代码后快速查看修改成果？

### DISCUSSION THREAD

**Discussion by Andy-AO** [bookfere/Ebook-Translator-Calibre-Plugin#260] [General]
想为插件添加功能，有个地方不懂。
修改代码之后如何运行这个插件查看结果呢？
普通的Python程序都可以直接在IDE上运行，可是这个程序不行。
显而易见的运行方法是把程序打包，然后在首选项中导入插件，之后再运行查看结果。
可是这个过程有点太复杂了，这使得修改和查看结果的循环频率很低。
平常你们是怎么操作的？还望指教，谢谢。

**Comment by Andy-AO** [comment]
我在文档上看到了，可以使用`calibre-debug -s; calibre-customize -b /path/to/your/plugin/folder; calibre`

**Comment by GoTop** [comment]
> 我在文档上看到了，可以使用`calibre-debug -s; calibre-customize -b /path/to/your/plugin/folder; calibre`

请问是在哪个文档看到这段代码的，我也想试试编辑一下calibre的插件，但是不知道怎么调试插件的代码

**Comment by Andy-AO** [comment]
@GoTop 在这里
https://manual.calibre-ebook.com/creating_plugins.html#debugging-plugins

**Comment by bookfere** [comment]
调试 GUI 插件要添加一个选项 `-g`，否则会进入 Calibre Shell 交互模式：

```
calibre-debug -s; calibre-customize -b /path/to/your/plugin/folder; calibre -g
```
